src/category_theory/limits/shapes/finite_limits.lean

```diff
@@ -39,9 +39,14 @@ instance has_limits_of_shape_of_has_finite_limits
   (J : Type v) [small_category J] [fin_category J] [has_finite_limits C] :
   has_limits_of_shape J C := has_finite_limits.out J
 
+@[priority 100]
+instance has_finite_limits_of_has_limits_of_size [has_limits_of_size.{v' u'} C] :
+  has_finite_limits C :=
+⟨λ J hJ hJ', by { haveI := has_limits_of_size_shrink.{0 0} C,
+  exact has_limits_of_shape_of_equivalence (fin_category.equiv_as_type J) }⟩
```

Comment:
Having limits of any size guarantees the existence of finite limits.
This unlocks a lot of the shapes API when we have some instance of `has_limits_of_size.{v' u'} C`,
but otherwise the analogous instances needs to be added manually for each shape.
(For example `C => D` has <insert shape> when `D` has them).